h/views/accounts.py

```diff
@@ -551,8 +559,10 @@ def post(self):
         )
 
     def _update_notifications(self, appstruct):
-        for notification in self._user_notifications():
-            notification.active = notification.type in appstruct["notifications"]
+        self.subscription_svc.set_subscriptions(
+            user_id=self.request.authenticated_userid,
+            subscriptions={type_: True for type_ in appstruct["notifications"]},
+        )
```

Comment:
If this is basically the only place we want this, and it want's to communicate in sets, perhaps we should too? This would prevent unnecessary mapping.